### PR TITLE
feat: message verification

### PR DESCRIPTION
**Pull request by ItsANameToo**
## Proposed changes

Adds message verification and message signing / verification tests.
Based on the verify functionality implemented by @supaiku0 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [x] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/docs/contributing) documentation
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by faustbrian** [comment]
@ItsANameToo there have been some changes in the last commits that should make it possible to finish this now.

**Comment by supaiku0** [comment]
@ItsANameToo  Before you commit please also move `Message.ex` to`lib/arkcosystem/crypto`.

File: lib/utils/ec_key.ex [hunk 1]
```diff
@@ -44,6 +44,15 @@ defmodule ArkEcosystem.Crypto.Utils.EcKey do
     |> private_key_to_address(network_address)
   end
 
+  def pubkey_from_hex(public_key) do
+    BtcCore.decompress(public_key)
+  end
+
+  def verify_message(message, signature, public_key) do
```
Comment (faustbrian) [inline lib/utils/ec_key.ex]: This method should be inside `message.ex` as it is specific to the message entity.

File: lib/utils/ec_key.ex [hunk 2]
```diff
@@ -44,6 +44,15 @@ defmodule ArkEcosystem.Crypto.Utils.EcKey do
     |> private_key_to_address(network_address)
   end
 
+  def pubkey_from_hex(public_key) do
```
Comment (faustbrian) [inline lib/utils/ec_key.ex]: This method should be named `public_key_from_hex`. Abbreviations don't really offer any benefit here.

**Comment by ItsANameToo** [comment]
@supaiku0 will do :)

**Comment by ItsANameToo** [comment]
@faustbrian you can take a look again

**Pull request merged**